src/SDK.ts

```diff
@@ -696,55 +786,74 @@ export default class SDK implements ISDK {
     const axiosInstance = axios.create();
     axiosRetry(axiosInstance, { retries: this.configuration.maxRequestRetriesOnFailure });
 
-    const { authenticatedUserToken, isReconnectChat, isPersistentChat, chatId} = sessionCloseOptionalParams;
+    const { authenticatedUserToken, isReconnectChat, isPersistentChat, chatId } = sessionCloseOptionalParams;
 
     const headers: StringMap = Constants.defaultHeaders;
     const data: any = {}; // eslint-disable-line @typescript-eslint/no-explicit-any
     data.chatId = chatId;
 
     if (authenticatedUserToken) {
-      endpoint = `${this.omnichannelConfiguration.orgUrl}/${OmnichannelEndpoints.LiveChatAuthSessionClosePath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}?channelId=${this.omnichannelConfiguration.channelId}`;
+      requestPath = `${this.omnichannelConfiguration.orgUrl}/${OmnichannelEndpoints.LiveChatAuthSessionClosePath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}?channelId=${this.omnichannelConfiguration.channelId}`;
       headers[OmnichannelHTTPHeaders.authenticatedUserToken] = authenticatedUserToken;
     }
 
     if (isReconnectChat) {
-      endpoint += `&isReconnectChat=true`;
+      requestPath += `&isReconnectChat=true`;
     }
 
-    if (isPersistentChat){
-      endpoint += `&isPersistentChat=true`;
+    if (isPersistentChat) {
+      requestPath += `&isPersistentChat=true`;
     }
 
+    const url = `${this.omnichannelConfiguration.orgUrl}${requestPath}`;
+    const method = "POST";
     const options: AxiosRequestConfig = {
       data,
       headers,
-      method: "POST",
-      url: endpoint
+      method,
+      url
     };
 
     return new Promise(async (resolve, reject) => {
       try {
         const response = await axiosInstance(options);
         const elapsedTimeInMilliseconds = timer.milliSecondsElapsed;
         if (this.logger) {
-          this.logger.log(LogLevel.INFO,
-            OCSDKTelemetryEvent.SESSIONCLOSESUCCEEDED,
-            { RequestId: requestId, Region: response.data.Region, ElapsedTimeInMilliseconds: elapsedTimeInMilliseconds, TransactionId: response.headers["transaction-id"] },
-            "Session Close Succeeded");
+
+          const customData = {
+            RequestId: requestId,
+            Region: response.data.Region,
+            ElapsedTimeInMilliseconds: elapsedTimeInMilliseconds,
+            TransactionId: response.headers[Constants.transactionid],
+            RequestPath: requestPath,
+            RequestMethod: method,
+            ResponseStatusCode: response.status
+          };
+
+          const description = "Session Close succeeded";
+          this.logger.log(LogLevel.INFO, OCSDKTelemetryEvent.SESSIONCLOSESUCCEEDED, customData, description);
         }
```

Comment:
can we evaluate the possibility of abstract this blocks?, I see it all around with minimal diffs among implementations , if we could find a way , this will reduce the code significatively

Reply:
Found a way and changed the code